src/pages/TodosPage.jsx

```diff
@@ -0,0 +1,118 @@
+import { useSearchParams, useNavigate } from 'react-router';
```

Comment:
There is a lot going on in this component. Some of it doesn't directly pertain to the `todoPage` component but rather to `TodoList`. It's a good idea to separate the logic here into both components.